### PR TITLE
Support wav audio play on Windows

### PR DESCRIPTION
**Pull request by timlyeee**
re : https://github.com/cocos/3d-tasks/issues/9083

### CODE REVIEW

**Comment by timlyeee** [comment]
@cocos-robot run test cases

**Review by dumganhar — APPROVED** [review]

File: native/cocos/audio/common/utils/include/primitives.h [hunk 1]
```diff
@@ -14,13 +14,17 @@
...
  */
 
-#ifndef COCOS_AUDIO_PRIMITIVES_H
-#define COCOS_AUDIO_PRIMITIVES_H
+#pragma once
+#if CC_PLATFORM == CC_PLATFORM_ANDROID
+    #include <sys/cdefs.h>
+#elif CC_PLATFORM == CC_PLATFORM_WINDOWS
+    #include <sys/types.h>
+    #include "audio/include/AudioMacros.h"
```
Comment (dumganhar) [inline native/cocos/audio/common/utils/include/primitives.h]: Why need to include `AudioMacros.h` on Windows?

File: native/cocos/audio/common/utils/primitives.cpp [hunk 2]
```diff
@@ -14,28 +14,34 @@
...
 
-#include "audio/android/audio_utils/include/audio_utils/primitives.h"
-#include "audio/android/audio_utils/private/private.h"
-#include "audio/android/cutils/bitops.h" /* for popcount() */
+#include "audio/common/utils/include/primitives.h"
+#include "audio/common/utils/private/private.h"
+#if CC_PLATFORM == CC_PLATFORM_ANDROID
+    #include "audio/android/cutils/bitops.h" /* for popcount() */
+#else
+    #include <cocos/base/Utils.h>
```
Comment (dumganhar) [inline native/cocos/audio/common/utils/primitives.cpp]: "base/Utils.h"

Comment (timlyeee) [reply native/cocos/audio/common/utils/primitives.cpp]: 1

File: native/cocos/audio/common/utils/include/primitives.h (hunk 1 above)
Comment (timlyeee) [reply native/cocos/audio/common/utils/include/primitives.h]: To use __BEGIN_DECLS marco as well as android.

File: native/cocos/audio/include/AudioMacros.h [hunk 3]
```diff
@@ -75,3 +75,7 @@
         CC_LOG_DEBUG("(" QUOTEME(condition) ") failed, message: " fmt, ##__VA_ARGS__); \
         break;                                                                         \
     }
+#ifndef __BEGIN_DECLS
```
Comment (PatriceJiang) [inline native/cocos/audio/include/AudioMacros.h]: `__BEGIN_DECLS` is a built-in macro, we should define our own version.

Comment (timlyeee) [reply native/cocos/audio/include/AudioMacros.h]: 1

**Review by PatriceJiang — APPROVED** [review]

**Review by dumganhar — APPROVED** [review]

**Pull request merged**